### DISCUSSION TITLE
Custom misconfiguration rules uses `--policy` flag instead of `--config-policy`

### DISCUSSION THREAD

**Discussion by jrarmstro** [aquasecurity/trivy#5759] [Documentation]
### Description

The custom policy docs for misconfiguration reference the `--policy` flag, when in reality the `--config-policy` flag should be used.

The `--namespaces` flag is also used, whereas `trivy config --help` mentions `--policy-namespaces`.

### Link

https://aquasecurity.github.io/trivy/v0.48/docs/scanner/misconfiguration/custom/

### Suggestions

Use the correct flags in the documentation

**Comment by nikpivkin** [comment]
Hi @jrarmstro !

`policy` is an alias for `config-policy `. `namespaces` is an alias for `policy-namespaces `.

@simar7 Should the help command print aliases?

**Reply by jrarmstro** [reply]
I've been working through this and recognized that to be the case. I think the docs should align with `--help`, regardless of whether `--help` prints aliases.

My issue seems to be with the subtypes specifier. Still debugging.

**Reply by jrarmstro** [reply]
The documentation at https://aquasecurity.github.io/trivy/v0.48/docs/scanner/misconfiguration/custom/selectors/ talks about the `service:` subtype field, but doesn't seem to link to the available services - I know I had found a list somewhere earlier, though.

**Reply by jrarmstro** [reply]
I'll move forward without defining a subtype. Improving the documentation (or support?) for subtypes would probably help folks in the future. Thanks for the quick response.

**Reply by simar7** [reply]
> The documentation at https://aquasecurity.github.io/trivy/v0.48/docs/scanner/misconfiguration/custom/selectors/ talks about the `service:` subtype field, but doesn't seem to link to the available services - I know I had found a list somewhere earlier, though.

The services are listed [here](https://avd.aquasec.com/misconfig/aws/) but you're right we can probably link this page in the docs for visibility.

**Reply by simar7** [reply]
> Hi @jrarmstro !
> 
> 
> 
> `policy` is an alias for `config-policy `. `namespaces` is an alias for `policy-namespaces `.
> 
> 
> 
> @simar7 Should the help command print aliases?
> 
> 

I think printing aliases is a general question we can ask at the Trivy level. 

Personally I'm fine with showing aliases but it might be too verbose. 

@knqyf263  what do you think?

**Reply by knqyf263** [reply]
AFAIK, cobra (technically pflag) doesn't have the functionality to show aliases.
https://github.com/spf13/pflag?tab=readme-ov-file#mutating-or-normalizing-flag-names
